src/Sentry.AspNetCore/SentryWebHostBuilderExtensions.cs

```diff
@@ -60,6 +60,9 @@ public static IWebHostBuilder UseSentry(this IWebHostBuilder builder, string dsn
                     o.Dsn = new Dsn(aspnetOptions.Dsn);
                 }
 
+                o.Environment = aspnetOptions.Environment ??
+                                Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
```

Comment:
This environment variable isn't guaranteed to be set. The value you get from the hosting environment currently [defaults to `Production`](https://github.com/aspnet/Hosting/blob/8836de411408257dd75f5db5f7d6724bfb0fab9e/src/Microsoft.AspNetCore.Hosting/Internal/HostingEnvironment.cs#L10) and is then overridden based on either the `launchSettings.json` (usually set to `Development`) or the system environment.

What I tried was removing this line and then modifying the `ConfigureLogging` call below it like so
```diff
 builder.ConfigureLogging((context, logging) =>
 {
     context.Configuration.GetSection("Sentry").Bind(aspnetOptions);
 
+    if (string.IsNullOrEmpty(aspnetOptions.Environment))
+    {
+        aspnetOptions.Environment = context.HostingEnvironment.EnvironmentName;
+    }
 
     configureOptions?.Invoke(aspnetOptions);
 
     logging.AddFilter<SentryLoggerProvider>(
         "Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware",
         LogLevel.None);
 
     logging.AddSentry(o => aspnetOptions.Apply(o));
 });
```

In my quick testing it works, but I'm not sure how "correct" it is to do it like that.

Reply:
True that when the value is not set is should default to Production
that's why usually it is followed by ?? "Production" which I forgot to add
but adding value to launchSettings.json for example would work as it sets the process environment variable.

In any case, what you just proposed is a better solution. I'll make sure to use `HostingEnvironment` from context! 

Thanks again!